File: ios/LookSeeProto/Models/SubmissionModels.swift
//
//  SubmissionModels.swift
//  LookSeeProto
//
//  Created by Ian Thompson on 2/15/26.
//  Updated to include upload metadata (location + descriptions + userEmail + landmark tracking).
//

import Foundation

enum MediaKind: String, Codable {
    case video
    case photo
}

struct InitSubmissionRequest: Codable {
    let userEmail: String
    let label: String
    let mediaKind: MediaKind
    let filename: String
    let contentType: String
}

struct InitSubmissionResponse: Codable {
    let submissionId: String
    let uploadUrl: String
    let s3Key: String
}

// Sent on /submissions/complete after PUT upload succeeds.
struct CompleteSubmissionRequest: Codable {
    let submissionId: String
    let s3Key: String
    let userEmail: String

    let label: String
    let landmarkId: String?
    let landmarkLabel: String?
    let mediaKind: MediaKind

    let shortDescription: String?
    let userDescription: String?

    let latitude: Double?
    let longitude: Double?
    let horizontalAccuracy: Double?
}

// Returned locally after the positive submission has completed.
// This is not decoded directly from the backend.
struct PositiveSubmissionResult {
    let submissionId: String
    let landmarkId: String?
    let mediaKind: MediaKind
    let s3Key: String
}

File: Android/LookSee/app/src/main/java/looksee/angelll/com/models/SubmissionModels.kt
package looksee.angelll.com.models

import com.google.gson.annotations.SerializedName

/** The media categories accepted by the submissions API. */
enum class MediaKind {
    @SerializedName("video")
    VIDEO,

    @SerializedName("photo")
    PHOTO,
}

/** Request sent before uploading positive media to the returned URL. */
data class InitSubmissionRequest(
    val userEmail: String,
    val label: String,
    val landmarkId: String? = null,
    val mediaKind: MediaKind,
    val filename: String,
    val contentType: String,
)

/** Upload destination returned by the submission initialization endpoint. */
data class InitSubmissionResponse(
    val submissionId: String,
    val uploadUrl: String,
    val s3Key: String,
)

/** Request sent to /submissions/complete after the media PUT succeeds. */
data class CompleteSubmissionRequest(
    val submissionId: String,
    val s3Key: String,
    val userEmail: String,
    val label: String,
    val landmarkId: String?,
    val landmarkLabel: String?,
    val mediaKind: MediaKind,
    val shortDescription: String?,
    val userDescription: String?,
    val latitude: Double?,
    val longitude: Double?,
    val horizontalAccuracy: Double?,
)

/** Local result returned after a positive submission completes. */
data class PositiveSubmissionResult(
    val submissionId: String,
    val landmarkId: String?,
    val mediaKind: MediaKind,
    val s3Key: String,
)
